### PR TITLE
Add multi options possibility & rotate option

### PR DESCRIPTION
**Pull request by remy-theroux**
Voici une proposition pour pouvoir appliquer plusieurs options dans la transformation d'images.
C'est clairement un breaking change, au niveau de la configuration.
J'ai ajouté l'option rotate pour pouvoir tester du multi options.
Dites moi ce que vous en pensez, je peux ajuster.

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -45,30 +45,51 @@ public function getConfigTreeBuilder(): TreeBuilder
...
                                         ->end()
                                     ->end()
+                                    ->arrayNode('rotate')
+                                        ->children()
+                                            ->enumNode('angle')
+                                                ->values([0, 90, 180, 270])
+                                            ->end()
+                                        ->end()
+                                    ->end()
+                                    ->arrayNode('height')
```
Comment (remy-theroux) [inline src/DependencyInjection/Configuration.php:83]: Should be improved as all option that can be used in standalone can be accepted at root level (resize_type for example can be used alone in doc)

File: src/Option/OptionFactory.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Mezcalito\ImgproxyBundle\Option;
+
+use function Symfony\Component\String\u;
+
+final class OptionFactory
+{
+    public static function fromName(string $optionName, array $optionParams): OptionInterface
+    {
+        $fqcn = '\\Mezcalito\\ImgproxyBundle\\Option\\' . mb_ucfirst(mb_strtolower($optionName));
```
Comment (remy-theroux) [inline src/Option/OptionFactory.php]: Used class name in sync with option name to build options

File: src/Option/Rotate.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+namespace Mezcalito\ImgproxyBundle\Option;
+
+class Rotate extends Option
+{
+    public function getParts(): array
+    {
+        $parts = [];
+
+        if (!empty($this->params['angle'])) {
+            $parts['angle'] = $this->params['angle'];
```
Comment (remy-theroux) [inline src/Option/Rotate.php:23]: Params that are not usable as standalone are not option so use them as scalar values.

File: src/Resolver.php [hunk 4]
```diff
@@ -47,7 +49,11 @@ public function getBrowserPath(string $src, string $presetName): string
         }
 
         $preset = $this->presets[$presetName];
-        $options = (new Resize($preset['resize']))->resolve();
+        foreach ($preset['options'] as $optionName => $optionParams) {
+            $option = OptionFactory::fromName($optionName, $optionParams);
+            $options[] = $option->resolve();
+        }
+        $options = implode('/', $options);
```
Comment (remy-theroux) [inline src/Resolver.php]: Separator is configurable, we could let it be a config of the bundle

**Comment by remy-theroux** [comment]
@julienj @leoberry C'est une ébauche de changement de philosophie concernant les options même s'il manque encore quelques éléments

**Comment by julienj** [comment]
Hello Remy

Thank you for your contribution and for your patience. We wanted to let you know that our lead developer is currently unavailable and won't be back until the end of October. We will review your contribution thoroughly once they return.

Thank you again for your understanding!

**Comment by loicsapone** [comment]
Hi @remy-theroux, thanks for your work! 

I agree with you about the BC break and the fact that we need to find another way of validating the bundle configuration.
At first, we wanted to keep things close to LiipImagineBundle with filters (like thumbnails, crops, etc.) to make migration easier. But I think we need to move away from this model to stick a little more to the imgproxy model and let the user define sets to reuse a set of transformations.

For the moment, can you fix the CI (PHP-CS-Fixer and PHPUnit) so I can merge this PR

I'll try to work on the new functionality in the coming weeks

**Comment by remy-theroux** [comment]
Hi, @mdanet-gobocom will fix the MR. Happy to have contributed.

**Comment by mdanet-gobocom** [comment]
Hello !

Thank you for your feedback and the work provided which was valuable to us. 
I will happily collaborate again on this project.

**Review by loicsapone — APPROVED** [review]

**Pull request merged**